packages/abi-gen/src/index.ts

```diff
@@ -70,16 +70,26 @@ function registerPartials(partialsGlob: string): void {
     }
 }
 
-function writeOutputFile(name: string, renderedTsCode: string): void {
+function makeOutputFilePath(name: string): string {
     let fileName = toSnakeCase(name);
     if (fileName === 'z_r_x_token') {
         fileName = 'zrx_token';
     }
-    const filePath = `${args.output}/${fileName}.ts`;
+    return `${args.output}/${fileName}.ts`;
+}
+
+function writeOutputFile(name: string, renderedTsCode: string): void {
+    const filePath = makeOutputFilePath(name);
     fs.writeFileSync(filePath, renderedTsCode);
     logUtils.log(`Created: ${chalk.bold(filePath)}`);
 }
 
+function isOutFileUpToDate(abiFile: string, outFile: string): boolean {
+    const abiTime = fs.statSync(abiFile).mtimeMs;
```

Comment:
I like that. Maybe we can go even further and hash the ABI? That way if someone removes all the artifacts and recompiles the same contracts we still do no work.

Reply:
I do like that idea, and I saw that [that's how sol-compiler works](https://github.com/0xProject/0x-monorepo/blob/988a3347582cd812aadeaa0cdaef67389c8d833d/packages/sol-compiler/src/compiler.ts#L131).

But then the question is: where would we store the hash of the ABI?

In sol-compiler's case, the hash of the source (and it's dependencies! interestingly) is stored in the ABI output (as field `sourceTreeHashHex`).

In abi-gen's case, there is no existing, natural place to keep that hash.  Perhaps we could store it in a comment in the contract wrapper?  Or perhaps in some side file?  Those seemed convoluted / overly complicated to me, so I opted for the timestamp method.  And, personally, if I wipe out the ABI files and generate them anew, that means that I want to force regeneration of the wrappers too, though admittedly I'd only do that when I'm questioning my own sanity. :)

The more I think about it, the more I like the idea of putting the ABI hash in a comment in the generated wrapper.  But that would be more work, and if that's the route you'd prefer then I would advise we push through the timestamp changes and then have a separate piece of work to replace the timestamp checking with hashing.  In any case, you tell me what you want and I'll make it happen. :)